jina/peapods/runtimes/head/__init__.py

```diff
@@ -37,7 +37,12 @@ def __init__(
         """
         super().__init__(args, cancel_event, **kwargs)
 
+        if args.name is None:
+            args.name = ''
         self.name = args.name
+        self.pod_name = (
```

Comment:
isn't there a better way to handle this? What problem is it handling?

Reply:
I don't like it either but the problem is that we need to know the pod name in a k8s setting for the pool. The pool observes all changes in a k8s namespace, no matter it's position in the Flow. Then the head runtime needs to be able to send a message to a specific pod  which is identified by name. 
To make it less guess work, I could pass it down explicitly and not derive if from the name of the head?

Reply:
I would be more explicit yes

Reply:
I changed it now so that I use the env variable JINA_POD_NAME instead. This one is always set in K8s (unless the user provides their own template and removes it).
In a none K8s settings this value acually does not matter and it can be whatever static string, I default to worker in this case. But we happen to set the env variable in this case to sth else, it also will work.